src/test/java/com/rocketden/main/service/ProblemServiceTests.java

```diff
@@ -452,6 +452,30 @@ public void editProblemBadTestCaseOutput() {
 
         assertEquals(ProblemError.INVALID_INPUT, exception.getError());
     }
+    @Test
+    public void editProblemBadApproval() {
+        Problem problem = new Problem();
+        problem.setName(NAME);
+        problem.setDescription(DESCRIPTION);
+        problem.setDifficulty(ProblemDifficulty.MEDIUM);
+        problem.setOutputType(ProblemIOType.STRING);
+        ProblemInput problemInput = new ProblemInput(INPUT_NAME, IO_TYPE);
+        problem.addProblemInput(problemInput);
+
+        Mockito.doReturn(problem).when(repository).findProblemByProblemId(problem.getProblemId());
+
+        ProblemDto updatedProblem = ProblemMapper.toDto(problem);
+        updatedProblem.setApproval(true);
+
+        ArrayList<ProblemTestCaseDto> emptyCases = new ArrayList<ProblemTestCaseDto>();
```

Comment:
small nit but a suggestion would be to use List as the main type instead of ArrayList

```suggestion
        List<ProblemTestCaseDto> emptyCases = new ArrayList<ProblemTestCaseDto>();
```

Reply:
k